stl/inc/memory

```diff
@@ -1419,13 +1437,19 @@ template <class _Dx, class _Ty>
 _Dx* get_deleter(const shared_ptr<_Ty>&) noexcept = delete; // requires static RTTI
 #endif // _HAS_STATIC_RTTI
 
+enum class _Init_form : bool { _Default_init, _Value_init };
+
 // CLASS TEMPLATE _Ref_count_obj2
-template <class _Ty>
+template <class _Ty, _Init_form _Init>

```

Comment:
Wouldn't it be better to just introduce a new constructor taking some tag type? It's a polymorphic class so each instantiation needs to have it's own vtable. Ditto with all `_Ref_count_*`s.

Reply:
I agree it would be better. Let me see what I can come up with.